runtime/jvmti/suspendhelper.cpp

```diff
@@ -45,7 +45,15 @@ suspendThread(J9VMThread *currentThread, jthread thread, UDATA allowNull, UDATA
 				if (currentThread == targetThread) {
 					*currentThreadSuspended = TRUE;
 				} else {
-					currentThread->javaVM->internalVMFunctions->internalExitVMToJNI(currentThread);
+					J9InternalVMFunctions *vmFuncs = currentThread->javaVM->internalVMFunctions;
+					/* It is a well known problem that suspending a Java thread may cause deadlocks in the application if a resource is 
+					 * held while suspended. To prevent this happening in the Thread JCL library, check that the Object java.lang.Thread.lock 
+					 * has been released before suspending. To make sure of this, acquire the lock while halt is in progress.
```

Comment:
Can this comment point to the Thread.suspend java implementation?  And clarify that the other Thread state calls like isAlive, etc will block forever if we suspend while the thread is holding the Thead.lock?

Reply:
Also, the comment (and PR title) say exactly the opposite of what the code is doing - it's acquiring the lock around the suspension, not releasing it.

Reply:
I've updated the comments to make it more clear.